src/librustc/diagnostics.rs

```diff
@@ -1944,6 +1944,38 @@ you want. Example:
 ```
 "##,
 
+E0492: r##"
+A borrow of a constant containing interior mutability was attempted. Erroneous
+code example:
+
+```
+#![feature(const_fn)]
+
+use std::cell::Cell;
+
+const A: Cell<usize> = Cell::new(1);
+const B: &'static Cell<usize> = &A;
+// error: cannot borrow a constant which contains interior mutability, create
+//        a static instead
+
+// or:
+struct C { a: Cell<usize> }
+
+const D: C = C { a: Cell::new(1) };
+const E: &'static Cell<usize> = &D.a; // error
+
+// or:
+const F: &'static C = &D; // error
+
+fn main() {}
+```
+
+To solve this error, either use statics or avoid using `cell` type. If you want
```

Comment:
It'd be great to have an example of these solutions, e.g. using one of the atomic types.

It'd also be nice to have an explanation of *why* this is an error: a `const` represents a constant value that should never change. If one takes a `&` reference to the constant, then one is taking a pointer to some memory location containing the value, normally this is perfectly fine: most values can't be changed via a shared `&` pointer, but interior mutability would allow it. That is, a constant value could be mutated. On the other hand, a `static` is explicitly a single memory location, which can be mutated at will.